src/GraphQL/Internal/AST.hs

```diff
@@ -76,12 +130,12 @@ data OperationDefinition = Query    { getNode :: Node }
                          | Mutation { getNode :: Node }
                            deriving (Eq,Show)
 
-data Node = Node Name [VariableDefinition] [Directive] SelectionSet
+data Node = Node (Maybe Name) [VariableDefinition] [Directive] SelectionSet
```

Comment:
Node is one of those types that would benefit from named fields IMO. Why is the name `Maybe`? Should node creation fail if the name is invalid?